contribute/design/1.x/cstor-operator/doc.md

```diff
@@ -46,107 +46,154 @@ Following is the proposed CSPC schema in go struct:
-  DefaultRaidGroupType string `json:"defaultRaidGroupType"`
-
-  // OverProvisioning to enable over provisioning
-  // Optional -- defaults to false
-  OverProvisioning bool `json:"overProvisioning"`
-  // Compression to enable compression
-  // Optional -- defaults to off
-  // Possible values : lz, off
-  Compression string `json:"compression"`
+	// DataRaidGroupType is the data raid group raid type 
+	// Supported values are : stripe, mirror, raidz and raidz2
+	DataRaidGroupType string `json:"dataRaidGroupType"`
+
+	// WriteCacheRaidGroupType is the write cachce raid type 
+	// Supported values are : stripe, mirror, raidz and raidz2
+	WriteCacheRaidGroupType string `json:"writeCacheRaidGroupType"`
+
+	// ThickProvisioning to enable over thick provisioning
+	// Optional -- defaults to false
+	ThickProvisioning bool `json:"thickProvisioning"`
+
+	// Compression to enable compression
+	// Optional -- defaults to off
+	// Possible values : lz, off
+	Compression string `json:"compression"`
+
+	// Resources are the compute resources required by the cstor-pool
+	// container.
+	Resources *corev1.ResourceRequirements `json:"resources"`
+
+	// AuxResources are the compute resources required by the cstor-pool pod
+	// side car containers.
+	AuxResources *corev1.ResourceRequirements `json:"auxResources"`
+
+	// Tolerations, if specified, the pool pod's tolerations.
+	Tolerations []corev1.Toleration `json:"tolerations"`
+
+	// PriorityClassName if specified applies to this pool pod
+	// If left empty, DefaultPriorityClassName is applied.
+	// (See CStorPoolClusterSpec.DefaultPriorityClassName)
+	// If both are empty, not priority class is applied.
+	PriorityClassName string `json:"priorityClassName"`
 }
 
 // RaidGroup contains the details of a raid group for the pool
 type RaidGroup struct {
-  // Type is the raid group type
-  // Supported values are: stripe, mirror, raidz and raidz2
-
-  // stripe -- stripe is a raid group which divides data into blocks and
-  // spreads the data blocks across multiple block devices.
-
-  // mirror -- mirror is a raid group which does redundancy
-  // across multiple block devices.
-
-  // raidz -- RAID-Z is a data/parity distribution scheme like RAID-5, but uses dynamic stripe width.
-  // radiz2 -- TODO
-  // Optional -- defaults to `defaultRaidGroupType` present in `PoolConfig`
-  Type string `json:"type"`
-  // IsWriteCache is to enable this group as a write cache.
-  IsWriteCache bool `json:"isWriteCache"`
-  // IsSpare is to declare this group as spare which will be
-  // part of the pool that can be used if some block devices
-  // fail.
-  IsSpare bool `json:"isSpare"`
-  // IsReadCache is to enable this group as read cache.
-  IsReadCache bool `json:"isReadCache"`
-  // BlockDevices contains a list of block devices that
-  // constitute this raid group.
-  BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevices"`
+	BlockDevices []CStorPoolClusterBlockDevice `json:"blockDevices"`
```

Comment:
The BlockDevice here could be confused with BlockDevice used by the NDM. 
Also - CStorPoolClusterBlockDevice - sounds misleading, as this is BlockDeviceDetails of a cStorPool (not cluster wide). 

Just thinking out loud here: would it make sense to say `CSPBlockDevices []CStorPoolBlockDevice` and `json:cspBlockDevices`

Reply:
makes sense

Reply:
done.